### PR TITLE
chore: unify make targets for architectures

### PR DESCRIPTION
**Pull request by rbarabas**
Unifies make targets for the various architectures and upgrades FDB to 7.1.7 to support M1 natively.

### CODE REVIEW

File: scripts/install_build_deps.sh [hunk 1]
```diff
@@ -13,34 +14,83 @@
...
+case "${OS}" in
+	"Darwin")
+		FDB_PACKAGE_NAME="FoundationDB-${FDB_VERSION}_${ARCH}.pkg"
+		FDB_PACKAGE_PATH="$(mktemp -d)/${FDB_PACKAGE_NAME}"
+		curl --create-dirs -Lo "$FDB_PACKAGE_PATH" "https://tigrisdata-pub.s3.us-west-2.amazonaws.com/${FDB_PACKAGE_NAME}"
+		echo "$FDB_SHA  $FDB_PACKAGE_PATH" | shasum -a 512 -c
+		sudo installer -pkg "$FDB_PACKAGE_PATH" -target /
+		;;
+	"Linux")
+		sudo apt-get install -y protobuf-compiler
```
Comment (pboros) [inline scripts/install_build_deps.sh:89]: Duplicate protobuf compiler installation, also done above.

Comment (rbarabas) [reply scripts/install_build_deps.sh]: Great catch, fixed! Thanks!

File: docker/Dockerfile [hunk 2]
```diff
@@ -12,9 +12,18 @@
...
+FROM ubuntu:jammy-20220531 AS build
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    gcc \
+    git \
+    golang \
```
Comment (efirs) [inline docker/Dockerfile:24]: What is the golang version bundled with `ubuntu:jammy-20220531`? We need to make sure it's at least 1.18.

File: docker/Dockerfile [hunk 3]
```diff
@@ -12,9 +12,18 @@
...
-
-RUN apt-get update && apt-get install -y --no-install-recommends apt-transport-https sudo
+FROM ubuntu:jammy-20220531 AS build
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    gcc \
```
Comment (efirs) [inline docker/Dockerfile:22]: Do we need to install gcc?

File: docker/Dockerfile [hunk 4]
```diff
@@ -24,12 +33,11 @@ RUN sh /build/install_build_deps.sh
 COPY go.mod /build
 WORKDIR /build
 RUN go mod download
-
 COPY . /build
+#RUN --mount=type=cache,target=/root/.cache/go-build rm -f server/service && make bins
+RUN rm -f server/service && make bins
```
Comment (efirs) [inline docker/Dockerfile:38]: mounting cache here, significantly speeds up the build

Comment (rbarabas) [reply docker/Dockerfile:38]: Thanks for the feedback, make sense, assumed so! I'm trying to figure out what's braking build in that it cannot find the protoc helper and tried to disable the BUILDKIT which helped me to get around the error locally. Unfortunately, as we can see it did not solve the build issue here so I'll be reverting this change soon.

File: docker/Dockerfile (hunk 3 above)
Comment (rbarabas) [reply docker/Dockerfile:22]: Yeah, one of the builds failed without it. I don't have the exact location but if we remove it we should see where it fails in `make {run,test,local_test}`!

File: scripts/install_build_deps.sh [hunk 5]
```diff
@@ -1,4 +1,5 @@
-#!/bin/bash
+#!/usr/bin/env bash
+set -ex
```
[the remark this answers is not in the document]
Comment (rbarabas) [reply scripts/install_build_deps.sh:2]: Sure, let's move it there!

File: docker/Dockerfile (hunk 2 above)
Comment (rbarabas) [reply docker/Dockerfile:24]: 1.18.1

File: docker/Dockerfile (hunk 4 above)
Comment (rbarabas) [reply docker/Dockerfile]: I added it back, didn't really help here.

**Comment by himank** [comment]
Very cool!

**Review by himank — APPROVED** [review]

**Pull request merged**